### ISSUE TITLE
Planning very sensitive to vehicle vibrations, frequently fails.

### ISSUE THREAD

**Issue by josh-wende** [ApolloAuto/apollo#14568]
Using Apollo 7.0:

When driving autonomously, the vehicle very frequently stops because a planning cycle fails due to the Piecewise Jerk Nonlinear Speed Optimizer determining that the QP problem is infeasible. I have determined that this happens when the vehicle shakes, presumably since the IMU velocity/acceleration values jump around during shaking. I can't figure out why I am having this issue now when I never did before, any insight into what might be causing this is greatly appreciated.

**Comment by qwetqwe** [comment]
Maybe current initial acceleration exceed the maximum acceleration because of vibrations, which leads to Ipopt solve failed

**Comment by zhanglonggao** [comment]
Hello! I have encounterd the same issue, have you sloved it ?

> Using Apollo 7.0:
> 
> When driving autonomously, the vehicle very frequently stops because a planning cycle fails due to the Piecewise Jerk Nonlinear Speed Optimizer determining that the QP problem is infeasible. I have determined that this happens when the vehicle shakes, presumably since the IMU velocity/acceleration values jump around during shaking. I can't figure out why I am having this issue now when I never did before, any insight into what might be causing this is greatly appreciated.

**Comment by josh-wende** [comment]
I added some functions that specifically checked for infeasible starting conditions (either initial acceleration/velocity out of range, or the initial velocity plus acceleration combined with the maximum jerk not being enough to slow down without going over the max speed) and modifies the inputs to what can be feasible if so. It's a bit hacky, but it works. I found that just setting the initial acceleration to always be 0 for the solver made it a lot better too, without really affecting the behavior too much.

Hope this helps.

**Comment by zhanglonggao** [comment]
well, thanks for your reply. I'd like to try the second way first. May I ask how to set the initial acceleration to always be 0 ?



At 2024-03-30 01:42:59, "Josh Wende" ***@***.***> wrote:

I added some functions that specifically checked for infeasible starting conditions (either initial acceleration/velocity out of range, or the initial velocity plus acceleration combined with the maximum jerk not being enough to slow down without going over the max speed) and modifies the inputs to what can be feasible if so. It's a bit hacky, but it works. I found that just setting the initial acceleration to always be 0 for the solver made it a lot better too, without really affecting the behavior too much.

Hope this helps.
